src/function/aggregate/distributive/mode.cpp

```diff
@@ -9,81 +9,39 @@
 
 #include "duckdb/common/unordered_map.hpp"
 
-namespace duckdb {
+#include <functional>
 
-template <class T>
-struct ModeState {
-	unordered_map<T, size_t> *frequency_map;
-};
+namespace std {
 
-struct ModeFunction {
-	template <class STATE>
-	static void Initialize(STATE *state) {
-		state->frequency_map = nullptr;
+template <>
+struct hash<duckdb::interval_t> {
+	inline size_t operator()(const duckdb::interval_t &val) const {
+		return hash<int32_t> {}(val.days) ^ hash<int32_t> {}(val.months) ^ hash<int64_t> {}(val.micros);
 	}
+};
 
-	template <class INPUT_TYPE, class STATE, class OP>
-	static void Operation(STATE *state, FunctionData *bind_data, INPUT_TYPE *input, ValidityMask &mask, idx_t idx) {
-		if (!state->frequency_map) {
-			state->frequency_map = new unordered_map<INPUT_TYPE, size_t>();
-		}
-		(*state->frequency_map)[input[idx]]++;
+template <>
+struct hash<duckdb::hugeint_t> {
+	inline size_t operator()(const duckdb::hugeint_t &val) const {
+		return hash<int64_t> {}(val.upper) ^ hash<int64_t> {}(val.lower);
 	}
+};
 
-	template <class STATE, class OP>
-	static void Combine(STATE &source, STATE *target) {
-		if (!source.frequency_map) {
-			return;
-		}
-		if (!target->frequency_map) {
-			target->frequency_map = source.frequency_map;
-			source.frequency_map = nullptr;
-			return;
-		}
-		for (auto &val : *source.frequency_map) {
-			(*target->frequency_map)[val.first] += val.second;
-		}
-	}
```

Comment:
No combine anymore? Any reason for it not to run in parallel?

Reply:
Not sure what you mean - `Combine` is in my copy of the code. There is now only one `ModeFunction` `struct` so maybe you were looking at the old one?  I tried to make the diffs easy to read, but when you mash two almost identical pieces of code together it can confuse it?